src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -199,5 +199,277 @@ function (indicator_max::IndicatorMax)(u::AbstractArray{<:Any,3},
+    # Copy alpha values such that smoothing is indpedenent of the element access order
+    alpha_tmp .= alpha
+
+    # Loop over interfaces
+    for interface in eachinterface(dg, cache)
+      # Get neighboring element ids
+      left  = cache.interfaces.neighbor_ids[1, interface]
+      right = cache.interfaces.neighbor_ids[2, interface]
+
+      # Apply smoothing
+      alpha[left]  = max(alpha_tmp[left],  0.5 * alpha_tmp[right], alpha[left])
+      alpha[right] = max(alpha_tmp[right], 0.5 * alpha_tmp[left],  alpha[right])
+    end
+  end
+
+  return alpha
+end
+
+function (indicator_ann::IndicatorNeuralNetwork{NeuralNetworkRayHesthaven})(
+    u::AbstractArray{<:Any,3}, mesh, equations, dg::DGSEM, cache; kwargs...)
+  @unpack indicator_type, alpha_max, alpha_min, alpha_smooth, alpha_continuous, alpha_amr, variable, network = indicator_ann
+
+  @unpack alpha, alpha_tmp, indicator_threaded, neighbor_ids = indicator_ann.cache
+  # TODO: Taal refactor, when to `resize!` stuff changed possibly by AMR?
+  #       Shall we implement `resize!(semi::AbstractSemidiscretization, new_size)`
+  #       or just `resize!` whenever we call the relevant methods as we do now?
+  resize!(alpha, nelements(dg, cache))
+  if alpha_smooth
+    resize!(alpha_tmp, nelements(dg, cache))
+  end
+
+  c2e = zeros(Int, length(mesh.tree))
+  for element in eachelement(dg, cache)
+    c2e[cache.elements.cell_ids[element]] = element
+  end
+
+
+  @threaded for element in eachelement(dg, cache)
+    indicator  = indicator_threaded[Threads.threadid()]
+    cell_id = cache.elements.cell_ids[element]
+    #neighbor_ids = Vector{Int}(undef, 2)
+
+    for direction in eachdirection(mesh.tree)
+      if !has_any_neighbor(mesh.tree, cell_id, direction)
+        neighbor_ids[direction] = element_id
+      continue
+      end
+      if has_neighbor(mesh.tree, cell_id, direction)
+        neighbor_cell_id = mesh.tree.neighbor_ids[direction, cell_id]
+        if has_children(mesh.tree, neighbor_cell_id) # Cell has small neighbor
+          if direction == 1
+            neighbor_ids[direction] = c2e[mesh.tree.child_ids[2, neighbor_cell_id]]
+          else
+            neighbor_ids[direction] = c2e[mesh.tree.child_ids[1, neighbor_cell_id]]
+          end
+        else # Cell has same refinement level neighbor
+          neighbor_ids[direction] = c2e[neighbor_cell_id]
+        end
+      else # Cell is small and has large neighbor
+        parent_id = mesh.tree.parent_ids[cell_id]
+        neighbor_cell_id = mesh.tree.neighbor_ids[direction, parent_id]
+        neighbor_ids[direction] = c2e[neighbor_cell_id]
+      end
+    end
+
+    # Calculate indicator variables at Gauss-Lobatto nodes
+    for i in eachnode(dg)
+      u_local = get_node_vars(u, equations, dg, i, element)
+      indicator[i] = indicator_ann.variable(u_local, equations)
+    end
+
+
+    # Cell average and interface values of the cell
+    X2 = sum(indicator)/nnodes(dg)
```

Comment:
Apparently this is how Ray & Hesthaven did it in their paper(s), thus we'll leave it as it is for now. @HenrZu this might be something you can try as well, although I am not sure how much impact this will have.